src/stan/services/util/gq_writer.hpp

```diff
@@ -81,7 +81,6 @@ class gq_writer {
       if (ss.str().length() > 0)
         logger_.info(ss);
       logger_.info(e.what());
-      return;
```

Comment:
If this isn't going to return, then the following `logger_` call outside the catch duplicates the `ss` message.  The full context is this:

```cpp
    try {
      model.write_array(rng, const_cast<std::vector<double>&>(draw), params_i,
                        values, false, true, &ss);
    } catch (const std::exception& e) {
      if (ss.str().length() > 0)
        logger_.info(ss);
      logger_.info(e.what());
    }
    if (ss.str().length() > 0)
      logger_.info(ss);
```

The conditional logging after the catch should be moved into the try block to avoid duplicating `logger_.info(ss)`.   

This is also making me think the logger itself should be ignoring empty string messages rather than duplicating this everywhere.

Reply:
Good catch.

Ignoring empty strings has a (nearly 5 year old!) issue open: https://github.com/stan-dev/stan/issues/2577